src/order.py

```diff
@@ -1,169 +1,44 @@
-        sell_value = self.sell_amount / pow(10, stable_coins[self.sell_token])
-        return buy_value > sell_value
-
-    def tokens(self) -> Set[Address]:
-        return {self.buy_token, self.sell_token}
-
-    def opposite_kind(self) -> String:
-        if self.kind == "sell":
-            return "buy"
-        return "sell"
-
-    def get_fee(self) -> Optional[int]:
-        if self.kind in {"buy", "sell"}:
-            url = "{0}fee?sellToken={1}&buyToken={2}&amount={3}&kind={4}".format(
-                base_url, self.sell_token, self.buy_token, str(self.sell_amount),
-                self.kind)
-            fee_response = requests.get(url).json()
-        else:
-            # TODO - log something here.
-            return
-        return int(fee_response["amount"]) or 0
-
-    def to_json(self, signed_message, public_key):
-        # TODO - make this less ugly
-        # self.__dict__ + extend by missing bits
-        json_str = '''{
-          "sellToken": "''' + str(self["sellToken"]) + '''",
-          "buyToken": "''' + str(self["buyToken"]) + '''",
-          "receiver": "''' + str(self["receiver"]) + '''",
-          "sellAmount": "''' + str(self["sellAmount"]) + '''",
-          "buyAmount": "''' + str(self["buyAmount"]) + '''",
-          "validTo": ''' + str(self["validTo"]) + ''',
-          "appData": "''' + str(self["appData"].hex()) + '''",
-          "feeAmount": "''' + str(self["feeAmount"]) + '''",
-          "kind": "''' + str(self["kind"]) + '''",
-          "partiallyFillable": ''' + str(self["partiallyFillable"]).lower() + ''',
-          "signature": "''' + str(signed_message.signature.hex()) + '''",
-          "signingScheme": "ethsign",
-          "sellTokenBalance": "''' + str(self["sellTokenBalance"]) + '''",
-          "buyTokenBalance": "''' + str(self["buyTokenBalance"]) + '''",
-          "from": "''' + public_key + '''"
-        }'''
-        return json.loads(json_str)
-
-    def place_order(self):
-        self.fee_amount = self.get_fee()
-        order_bytes = self.signable_bytes(domain)
-        order_hash = w3.keccak(order_bytes)
-        message = encode_defunct(primitive=order_hash)
-        signed_message = w3.eth.account.sign_message(message, private_key=private_key)
-        # Build Json out of Order + signed_message.signature.hex() + public_address
-        order_json = self.to_json(signed_message, public_address)
-        r = requests.post(base_url + "orders", json=order_json)
-        # TODO - log status code.
-
-    def counter_order(self):
-        # TODO - stop accessing fields by string keys.
-        res = Order(
-            sell_token=self.buy_token,
-            buy_token=self.sell_token,
-            receiver=public_address,
-            sell_amount=self.buy_amount,
-            buy_amount=self.sell_amount,
-            # 2 minutes in the future.
-            valid_to=int(time.time()) + 120,
-            kind=self.opposite_kind(),
-        )
```

Comment:
I feel like we are losing some useful methods here... `counter_order`, `opposite_kind`, `tokens`, `is_profitable`

Reply:
I only made the files to make an order via the API. I have other files where counter_order, opposite_kind etc are made but are under construction.

Reply:
That's nice to know, but could we at least put some of this lost matter somewhere so that it's not completely gone from the repo? Maybe leave them commented out with a todo saying reintroduce these.